Cython/Compiler/ExprNodes.py

```diff
@@ -610,6 +610,25 @@ def compile_time_value_error(self, e):
         error(self.pos, "Error in compile-time expression: %s: %s" % (
             e.__class__.__name__, e))
 
+    def as_exception_value(self, env):
+        # Return the compile_time_value if possible.
+        # This can be either a Python constant or a string
+        # for types that can't be represented by a Python constant
+        # (e.g. enums)
+        result = None
+        with local_errors(ignore=True):
+            result = self.compile_time_value(env)
+        if isinstance(result, Symtab.Entry):
+            result = result.cname
```

Comment:
`compile_time_value()` is used at parse time, which is long over here. This is called during the analysis phase. It seems better to use `self.entry` and the constant result value.